### PR TITLE
Auto instrumentation java

### PR DESCRIPTION
**Pull request by owais**

### CODE REVIEW

File: kuttl-test.yaml [hunk 1]
```diff
@@ -12,4 +12,5 @@ commands:
   - command: sleep 5s
 testDirs:
   - ./tests/e2e/
-timeout: 150
\ No newline at end of file
+#timeout: 150
+timeout: 30
```
Comment (dmitryax) [inline kuttl-test.yaml:16]: unrelated change?

File: internal/webhooks/handler.go [hunk 2]
```diff
@@ -29,13 +29,19 @@ const (
...
+	javaJVMArgument   = " -javaagent:/splunk/splunk-otel-javaagent-all.jar"
+	exporterOTLP      = "otlp"
+	exporterJaeger    = "jaeger-thrift-splunk"
 
-	annotationConfInjectionEnabled = "o11y.splunk.com/inject-config"
-	annotationStatus               = "o11y.splunk.com/injection-status"
-	annotationReason               = "o11y.splunk.com/injection-reason"
+	annotationInstrumentation = "o11y.splunk.com/instrumentation"
+	annotationStatus          = "o11y.splunk.com/status"
+	annotationReason          = "o11y.splunk.com/reason"
```
Comment (dmitryax) [inline internal/webhooks/handler.go:42]: Why `inject(ion)` term removed? I think it's pretty important part defining annotations purpose .

What about:
`o11y.splunk.com/inject: config`
`o11y.splunk.com/inject: java-agent`
or
`o11y.splunk.com/inject-instrumentation: config`
`o11y.splunk.com/inject-instrumentation: java-agent`

File: internal/webhooks/handler.go [hunk 3]
```diff
@@ -85,7 +91,24 @@ func (h *handler) Handle(ctx context.Context, req admission.Request) admission.R
...
+
+	instrumentation, ok := pod.Annotations[annotationInstrumentation]
+	if !ok {
+		return admission.Allowed("")
+	}
+
+	switch strings.ToLower(strings.TrimSpace(instrumentation)) {
+	case "config":
+		injectFn = h.injectConfig
+	case "java":
```
[the remark this answers is not in the document]
Comment (owais) [reply internal/webhooks/handler.go:104]: The key is `o11y.splunk.com/instrumentation` and value is `java`. Not sure if `java-agent` makes it any better.

File: internal/webhooks/handler.go (hunk 2 above)
Comment (owais) [reply internal/webhooks/handler.go:42]: IMO feel `instrumentation` is clear enough and won't cause any ambiguity or confusion but not married to it. can change now or later.

File: internal/webhooks/handler.go [hunk 4]
```diff
@@ -126,14 +149,62 @@ func configFromSpec(spec *v1alpha1.SplunkOtelAgentSpec) config {
 		}
 	}
 
+	cfg.javaImage = fmt.Sprintf(
```
[the remark this answers is not in the document]
Comment (owais) [reply internal/webhooks/handler.go]: Done

File: internal/webhooks/handler.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (dmitryax) [reply internal/webhooks/handler.go]: Ok, sounds good

File: internal/webhooks/handler.go (hunk 2 above)
Comment (dmitryax) [reply internal/webhooks/handler.go:42]: What is the attribute used in opentelemetry-collector-operator?

Comment (owais) [reply internal/webhooks/handler.go]: Changed it to `o11y.splunk.com/inject-java: true`. This is similar to what community operator does. I think this addresses your concern and also allows users to inject multiple instrumentations for any possible edge cases.

File: internal/webhooks/handler.go [hunk 5]
```diff
@@ -29,32 +29,46 @@ const (
...
+	volumeName        = "splunk-instrumentation"
+	initContainerName = "splunk-instrumentation"
+	javaJVMArgument   = " -javaagent:/splunk/splunk-otel-javaagent-all.jar"
+	exporterOTLP      = "otlp"
+	exporterJaeger    = "jaeger-thrift-splunk"
+
+	annotationJava   = "o11y.splunk.com/inject-java"
+	annotationConfig = "o11y.splunk.com/inject-config"
+	annotationStatus = "o11y.splunk.com/status"
+	annotationReason = "o11y.splunk.com/reason"
```
Comment (dmitryax) [inline internal/webhooks/handler.go:43]: I see `o11y.splunk.com/injection-<status|reason>` is renamed to `o11y.splunk.com/<status|reason>` but this PR doesn't rely on it. What is the reasoning for that? Do we want to reuse this attribute for reporting any other statuses/reasons?

Also looks like we report only `"error"` status. Shouldn't we report successfully injected events?

And using "o11y.splunk.com/reason" attribute for reporting error messages doesn't seem apparent to me.

Maybe we can use?
`o11y.splunk.com/inject-java/status` always reported if `o11y.splunk.com/inject-java=true`
`o11y.splunk.com/inject-java/error` added only if `o11y.splunk.com/inject-java/status=error`

Comment (owais) [reply internal/webhooks/handler.go]: Renamed it to fit with the other annotation but we renamed that to `inject-*` so renaming these back to `injection-status/reason`. Also added `success` annotation when there are no errors. We can elaborate this later and also emit k8s events in case of success/.failure.

**Review by dmitryax — APPROVED** [review]
